From 6e87f7263899d883833b4851b67efa2291c011ac Mon Sep 17 00:00:00 2001
From: AKASHI Takahiro <takahiro.akashi@linaro.org>
Date: Fri, 22 Jan 2021 10:43:49 +0900
Subject: [PATCH 1/1] tools: mkeficapsule: fill reserved members of structure

Fill reserved members of efi_firmware_management_capsule_image_header
structure with zero's for safety.

Signed-off-by: AKASHI Takahiro <takahiro.akashi@linaro.org>
Fixes: CID 316354
Signed-off-by: Heinrich Schuchardt <xypron.glpk@gmx.de>
---
 tools/mkeficapsule.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/tools/mkeficapsule.c b/tools/mkeficapsule.c
index 270943fc90..36f693f096 100644
--- a/tools/mkeficapsule.c
+++ b/tools/mkeficapsule.c
@@ -310,6 +310,9 @@ static int create_fwbin(char *path, char *bin, efi_guid_t *guid,
 	image.version = 0x00000003;
 	memcpy(&image.update_image_type_id, guid, sizeof(*guid));
 	image.update_image_index = index;
+	image.reserved[0] = 0;
+	image.reserved[1] = 0;
+	image.reserved[2] = 0;
 	image.update_image_size = bin_stat.st_size;
 	image.update_vendor_code_size = 0; /* none */
 	image.update_hardware_instance = instance;
-- 
2.29.2

